src/main/java/com/github/technus/tectech/thing/metaTileEntity/multi/GT_MetaTileEntity_TM_teslaCoil.java

```diff
@@ -100,35 +100,45 @@
-            GT_MetaTileEntity_TM_teslaCoil::addCapacitorToMachineList,
-            GT_MetaTileEntity_TM_teslaCoil::addFrameToMachineList);
-    private static final short[] casingTextures = new short[]{(texturePage << 7) + 16 + 6, 0};
-    private static final Block[] blockTypeFallback = new Block[]{sBlockCasingsBA0, null};
-    private static final byte[] blockMetaFallback = new byte[]{6, 0};
     private static final String[] description = new String[]{
             EnumChatFormatting.AQUA + translateToLocal("tt.keyphrase.Hint_Details") + ":",
             translateToLocal("gt.blockmachines.multimachine.tm.teslaCoil.hint.0"),//1 - Classic Hatches, Capacitor Hatches or Tesla Base Casing
             translateToLocal("gt.blockmachines.multimachine.tm.teslaCoil.hint.1"),//2 - Titanium Frames
     };
+
+    private static final IStructureDefinition<GT_MetaTileEntity_TM_teslaCoil> STRUCTURE_DEFINITION =
+            StructureDefinition.<GT_MetaTileEntity_TM_teslaCoil>builder()
+            .addShape("main", transpose(new String[][]{
+                    {"       ","       ","  BBB  ","  BBB  ","  BBB  ","       ","       "},
+                    {"       ","  BBB  "," BBBBB "," BBBBB "," BBBBB ","  BBB  ","       "},
+                    {"       ","  BBB  "," BBBBB "," BBBBB "," BBBBB ","  BBB  ","       "},
+                    {"       ","  BBB  "," BBBBB "," BBBBB "," BBBBB ","  BBB  ","       "},
+                    {"       ","       ","  BBB  ","  BCB  ","  BBB  ","       ","       "},
+                    {"       ","       ","       ","   C   ","       ","       ","       "},
+                    {"       ","  BBB  "," B F B "," BFCFB "," B F B ","  BBB  ","       "},
+                    {"       ","       ","       ","   C   ","       ","       ","       "},
+                    {"       ","  BBB  "," B F B "," BFCFB "," B F B ","  BBB  ","       "},
+                    {"       ","       ","       ","   C   ","       ","       ","       "},
+                    {"       ","  BBB  "," B F B "," BFCFB "," B F B ","  BBB  ","       "},
+                    {"       ","       ","       ","   C   ","       ","       ","       "},
+                    {"       ","  BBB  "," B F B "," BFCFB "," B F B ","  BBB  ","       "},
+                    {"       ","       ","       ","   C   ","       ","       ","       "},
+                    {"       ","       ","       ","   C   ","       ","       ","       "},
+                    {"       ","  DDD  "," D   D "," D C D "," D   D ","  DDD  ","       "},
+                    {" EE~EE ","EAAAAAE","EADDDAE","EADADAE","EADDDAE"," AAAAAE"," EEEEE "}
+            }))
+            .addElement('A', ofBlock(sBlockCasingsBA0, 6))
+            .addElement('B', ofBlock(sBlockCasingsBA0, 7))
+            .addElement('C', ofBlock(sBlockCasingsBA0, 8))
+            .addElement('D', lazy(t -> ofBlock(sBlockCasingsBA0, t.getCoilWindingMeta())))
```

Comment:
you can't use `lazy()` in this place it will fuck up the tesla tower, making all towers only allowed to use the small kind of coils.

Reply:
Yeah, `lazy()` will cache the returned `IStructureElement` for the entire lifetime of this `IStructureDefinition`. This is to reduce overhead of creating the same `IStructureElement` over and over when the block/item required to instantiate it is not yet known when `build()` is called, e.g. in fusion reactors.

`deferred` or `ofBlocksTiered` would be more appropriate here. `ofBlocksTiered` do require a more recent StructureLib though.